furusiya/model/components/fighter.py

```diff
@@ -1,15 +1,20 @@
 import colors
 import config
+from model.components.ai.base import AI
+from model.components.base import Component
 from model.game_object import GameObject
 from model.item import Item
 from main_interface import Game, message
 
 
-class Fighter:
+class Fighter(Component):
     """
     combat-related properties and methods (monster, player, NPC).
     """
-    def __init__(self, hp, defense, power, xp, weapon=None, death_function=None):
+    component_type = 'Fighter'
```

Comment:
I thought the component type is literally the type (metaclass). Why is this now specified as a string?